src/swell/utilities/geos.py

```diff
@@ -375,4 +375,62 @@ def resub(self, filename, pattern, replacement):
         with open(filename, 'w') as out_file:
             out_file.write(modified_text)
 
+    # --------------------------------------------------------------------------------------------------
+
+    def states_generator(self,
+                         background_frequency: str,
+                         window_length: str,
+                         window_begin_iso: str,
+                         model: str = 'geos_marine',
+                         marine_models: list = []
+                         ) -> list:
+
+        states = []
+        self.logger.info('Generating states for model: '+model)
+        if model == 'geos_ocean' or model == 'geos_marine':
+            states = self.marine_states(background_frequency, window_length, window_begin_iso,
+                                        marine_models)
+
+        return states
+
+    # --------------------------------------------------------------------------------------------------
+
+    def marine_states(self,
+                      background_frequency: str,
+                      window_length: str,
+                      window_begin_iso: str,
+                      marine_models: list
+                      ) -> list:
+
+        static_part = {"basename": "./", "read_from_file": 1}
+
+        # Calculate the number of states using background frequency and window length
+        number_of_states = int(isodate.parse_duration(window_length)
+                               / isodate.parse_duration(background_frequency)) + 1
+        self.logger.info('Number of states: ', number_of_states)
+
+        # Generate the list of states dictionary with date and marine filename entries.
+        # The date is calculated by adding the background frequency to the window begin date.
+        # The ocn_filename is calculated by adding the background frequency to the window begin date
+        states = []
+
+        # For FGAT and 4D-Var, the first state is the background state, hence we need to
+        # skip the first state in the loop by adding "-1" to the range function.
+        for i in range(number_of_states-1):
+            i += 1
```

Comment:
Range can take a `start` and `stop` argument. So, if you want to skip `i=0`, you can just do:

```suggestion
        for i in range(1, number_of_states):
```